app/src/main/java/nu/yona/app/YonaApplication.java

```diff
@@ -100,12 +102,27 @@ public void onCreate()
 		super.onCreate();
 		mContext = this;
 		eventChangeManager = new EventChangeManager();
-		sharedAppPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
-		sharedUserPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		validateAndInitializePreferences();
 		sharedAppDataState = eventChangeManager.getDataState();
 		Foreground.init(this);
 	}
 
+	private void validateAndInitializePreferences()
+	{
+		sharedAppPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		sharedUserPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
+		if (sharedUserPreferences.getString(AppConstant.SERVER_URL, null) != null)
+		{  // Need to exchange the preferences as corrective action, as a couple of previous builds had wrong(Exchanged Preferences) initializations.
+			nu.yona.app.utils.Logger.logi("Preferences", "Preferences Exchanged");
```

Comment:
This log message says "exchanged" (i.e. it's done now) while it's yet to exchange the preferences.
I see two options:
* Move this line to just below 122
* Copy this line to just below 122 and change this one into "Exchanging preferences"

Reply:
Done.